src/core/view.js

```diff
@@ -77,7 +78,7 @@ export class View {
         this.#resolveRenderPromise(undefined)
         delete this.renderPromise
       }
-    } else {
+    } else if (willRender) {
```

Comment:
I don't think it necessarily needs addressed here, but I find it confusing that we have `shouldRender` and `willRender` cases. It's not clear how they are different. If you *should* render, doesn't that imply that you *will*?

I think the easiest answer is probably to rename `willRender`. It really means that the view is being used in a context where rendering is not required. Or more specifically, it's being used to do all the other operations that normally accompany a page render, but after a frame update has already rendered.

But ideally we'd separate the responsibilities to avoid this sort of conditional check happening here. So maybe we're as well to stick with the existing language for now, and then look at refactoring in another PR which keeps the same behaviour?

Reply:
Yes, agreed to all of that 👍

Reply:
I agree the `shouldRender` / `willRender` is confusing here.

Is there a plan to refactor before releasing v8? If not then I reckon we should clarify `willRender` by renaming and adding a comment. e.g.:

```js
// A workaround to ignore tracked element mismatch reloads when performing a promoted Visit from a frame navigation

const shouldInvalidate = willRender
```

Reply:
I'd opened https://github.com/hotwired/turbo/pull/741 to limit the occurrence of `willRender`.

I'd also opened https://github.com/hotwired/turbo/pull/430 to try and introduce some abstractions for Frame navigation. They've both been kept up to date originally, but I eventually stopped dragging them forward.

Would either be worthwhile to rebase for inclusion in 8?